utils.go

```diff
@@ -115,12 +115,27 @@ loop:
 		select {
 		case workQueue <- i:
 			i++
-		case err = <-errCh:
-			break loop
+		case err = <-errCh: // abort pool if an error was discovered
+			if err != nil {
+				break loop
+			}
+		default:
+			time.Sleep(500 * time.Millisecond)
 		}
 	}
 
 	close(workQueue)
 	wg.Wait()
+	close(errCh)
+
+	if err != nil {
+		return results, err
+	}
+
+	for e := range errCh {
```

Comment:
If there's multiple errors in the channel, this will make it so that we return (only) the last one. That's intentional?

Reply:
yeah. unless we return an array of errors which is unconventional. what would you expect?

Reply:
I'd probably expect to see the first instead of the last error, but no big deal.